lib/hanami/config/assets.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require "dry/configurable"
+require "hanami/assets"
+
+module Hanami
+  class Config
+    # Hanami assets config
+    #
+    # This exposes all the settings from the standalone `Hanami::Assets` class, pre-configured with
+    # sensible defaults for actions within a full Hanami app. It also provides additional settings
+    # for further integration of actions with other full stack app components.
+    #
+    # @since 2.1.0
+    # @api public
+    class Assets
+      include Dry::Configurable
+
+      # @!attribute [rw] serve
+      #   Serve static assets.
+      #
+      #   When this is `true`, the app will serve static assets.
+      #
+      #   If not set, this will:
+      #
+      #     * Check if the `HANAMI_SERVE_ASSETS` environment variable is set to `"true"`.
+      #     * If not, it will check if the app is running in the `development` or `test` environment.
+      #
+      #   @example
+      #     config.assets.serve = true
+      #
+      #   @return [Hanami::Config::Actions::Cookies]
+      #
+      #   @api public
+      #   @since 2.1.0
+      setting :serve, default: Dry::Core::Constants::Undefined
```

Comment:
```suggestion
      setting :serve
```

I think we could honestly keep it simple here and leave this as the standard default to `nil`, especially given we're applying our proper default immediately upon initialization.

Otherwise, this would also be the only place where we use `Undefined` as a default setting value, and given the above, I don't think it's really justified here.